app/tables.py

```diff
@@ -5,7 +5,7 @@
 )
 from piccolo_api.session_auth.tables import SessionsBase
 
-DB = SQLiteEngine()
+DB = SQLiteEngine("/data/auth.sqlite")
```

Comment:
I think I made a mistake recommending this path.

If we try running this code outside of a container, it won't work, unless we add the `/data` folder at the root of our machines, which isn't great.

One solution is to put it in somewhere in `/var` folder like `/var/piccolo_admin/`. Or we just do something like:

```python
import os

DB_PATH = os.path.join(os.path.dirname(__file__), 'data')
DB = SQLiteEngine(DB_PATH)
```

I think this is my preferred solution.

And then our volume becomes:

```
VOLUME /code/app/data/
```

I think we should consider removing the `/code` folder. I'd rather just add everything to `/app` rather than `/code/app` - the `/code` prefix seems unnecessary, and we will have things other than code in there.

What do you think?

Reply:
> I think I made a mistake recommending this path.
> 
> If we try running this code outside of a container, it won't work, unless we add the `/data` folder at the root of our machines, which isn't great.

I just tried this in `virtualenv` and you are right. It doesn't work locally because there is no `data` directory

> One solution is to put it somewhere in `/var` folder like `/var/piccolo_admin/`. Or we just do something like:
> 
> ```python
> import os
> 
> DB_PATH = os.path.join(os.path.dirname(__file__), 'data')
> DB = SQLiteEngine(DB_PATH)
> ```

With these changes everything works in `virtualenv`, but with your changes the sqlite database with the name `data` (not the directory `data` with `auth.sqlite` inside) is created in the app directory which is fine for me, but then we have to leave `VOLUME /data`. If we need to create a data directory (to mount `VOLUME /app/data`) we need to do something like this

```python
import pathlib

DB_PATH = pathlib.Path("app/data/auth.sqlite").mkdir(parents=True)
```
Should the `data` be a directory or a file?

> I think this is my preferred solution.
> 
> And then our volume becomes:
> 
> ```
> VOLUME /code/app/data/
> ```
> 
> I think we should consider removing the `/code` folder. I'd rather just add everything to `/app` rather than `/code/app` - the `/code` prefix seems unnecessary, and we will have things other than code in there.
> 
> What do you think?

I tried rewriting the Dockerfile to something like this 

```dockerfile
FROM python:3.13-slim

WORKDIR /app

COPY ./requirements.txt /app/requirements.txt

COPY ./config.yaml /app/config.yaml

# VOLUME /data  or VOLUME /app/data # based on data is directory or file

RUN pip install --no-cache-dir -r /app/requirements.txt

COPY ./app /app

CMD ["python", "main.py"]
```